truss/templates/control/control/helpers/patch_applier.py

```diff
@@ -55,3 +60,44 @@ def _apply_model_code_patch(self, model_code_patch: ModelCodePatch):
                 filepath.unlink()
         else:
             raise ValueError(f"Unknown model code patch action {action}")
+
+    def _apply_python_requirement_patch(
+        self, python_requirement_patch: PythonRequirementPatch
+    ):
+        self._app_logger.debug(
+            f"Applying python requirement patch {python_requirement_patch.to_dict()}"
+        )
+        action = python_requirement_patch.action
+
+        if action == Action.REMOVE:
+            subprocess.run(
+                [
+                    self._pip_path,
+                    "uninstall",
+                    "-y",
+                    python_requirement_patch.requirement,
+                ],
+                check=True,
+            )
+
+        elif action == Action.UPDATE:
+            subprocess.run(
+                [
+                    self._pip_path,
+                    "install",
+                    python_requirement_patch.requirement,
+                ],
+                check=True,
+            )
+        else:
+            raise ValueError(f"Unknown python requirement patch action {action}")
+
+
+def _identify_pip_path() -> str:
+    if Path("/usr/local/bin/pip3").exists():
```

Comment:
does this way of identifying pip path break local workflows in a virtual environment?

Reply:
or is the assumption here that we only ever do this inside a Docker image?

Reply:
Yes, patch application only works on the docker image. This logic assumes a specific structure of docker image, e.g. it knows that the inference server uses the root python installation vs a virtual environment.